### PR TITLE
AJ-1591: startup validation

### PR DESCRIPTION
**Pull request by davidangb**
Some more prefactoring to help out eventually with #489 

In this PR:
* create a new "startup" package
* move InstanceInitializer* classes into the startup package
* expand our ConfigurationProperties classes to include the top-level `twds` property hierarchy and, importantly, the `twds.instance.workspace-id` property (which is populated from the `$WORKSPACE_ID` env var)
* add a new `StartupConfig` class in the startup package, which is responsible for performing some on-startup validations including validating the `twds.instance.workspace-id` property

NOT in this PR to save scope:
* I did not attempt to consolidate InstanceInitializer stuff into StartupConfig, but I think that would be possible if we wanted to do it
* I did not attempt to update any other classes that read from or require the `twds.instance.workspace-id` property. For instance, the `PfbQuartzJob` bean will fail if `$WORKSPACE_ID` is not a valid UUID, and I didn't try to change that
* I did not build out classes for ALL properties under `twds`; I figured we could do that as we need them

### CODE REVIEW

File: service/src/main/java/org/databiosphere/workspacedataservice/WorkspaceDataServiceApplication.java [hunk 1]
```diff
@@ -19,7 +17,6 @@
 @EnableRetry
 @EnableTransactionManagement
 @EnableCaching
-@EnableConfigurationProperties(WdsTenancyProperties.class)
```
Comment (davidangb) [inline service/src/main/java/org/databiosphere/workspacedataservice/WorkspaceDataServiceApplication.java:22]: I realized that by adding `@Configuration` to a property class, we don't need to list the class(es) here, so I removed it.

File: service/src/test/java/org/databiosphere/workspacedataservice/config/TenancyPropertiesDataPlaneTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package org.databiosphere.workspacedataservice.config;
```
Comment (davidangb) [inline service/src/test/java/org/databiosphere/workspacedataservice/config/TenancyPropertiesDataPlaneTest.java:1]: This was renamed from WdsTenancyPropertiesDataPlaneTest; unfortunately github is showing this as a delete of the old file and a brand new file

**Review by ashanhol — APPROVED** [review]
seems good, looks like just moving stuff around to accommodate the new startup/property validation

**Review by calypsomatic — APPROVED** [review]

File: service/src/main/java/org/databiosphere/workspacedataservice/startup/StartupConfig.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+/** */
+@Component
+public class StartupConfig {
+
+  private static final Logger logger = LoggerFactory.getLogger(StartupConfig.class);
+
+  private final TwdsProperties twdsProperties;
+
+  public StartupConfig(TwdsProperties twdsProperties) {
```
Comment (jladieu) [inline service/src/main/java/org/databiosphere/workspacedataservice/startup/StartupConfig.java:19]: I imagine `TwdsProperties` becoming a burrito of stuff and eventually not wanting to pull it all in here.

Is it possible to inject just the `TenancyProperties` and `InstanceProperties` directly here?

Comment (jladieu) [reply service/src/main/java/org/databiosphere/workspacedataservice/startup/StartupConfig.java:19]: (No changes needed here, asking for food for thought)

**Review by jladieu — APPROVED** [review]
One question (no changes needed, but food for thought), but LGTM!

File: service/src/main/java/org/databiosphere/workspacedataservice/startup/StartupConfig.java (hunk 3 above)
Comment (davidangb) [reply service/src/main/java/org/databiosphere/workspacedataservice/startup/StartupConfig.java:19]: The way I have it currently set up, it is not possible to inject `TenancyProperties` or `InstanceProperties`. To do this, we'd have to:
* remove `TenancyProperties` and `InstanceProperties` as member variables of `TwdsProperties`
* promote `TenancyProperties` and `InstanceProperties` to being first-class spring beans annotated with `@ConfigurationProperties(prefix = "twds.tenancy")` and `@ConfigurationProperties(prefix = "twds.instance")`, respectively

Then, any other bean that wants properties would have to inject whichever of `TwdsProperties`, `TenancyProperties`, and/or `InstanceProperties` that it needs

**Pull request merged**